### PR TITLE
2018Q3: Data Retrieval amendments

### PR DESCRIPTION
**Pull request by ajnavarro**
This is a continuation of #68 with amendments by @ajnavarro 
There is some previous discussion in #68.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -5,10 +5,18 @@
...
+        - [ ] Make indexes take less disk space
+        - [ ] Make indexes support more comparison expressions
 - [ ] Release a more stable gitbase version
+    - [ ] Make regression test
 - [ ] Engine using gitbase as backend
-- [ ] Change engine schema to match gitbase's
+    - [ ] Change engine schema to match gitbase's
+    - [ ] Replace actual JGit implementation with a query builder and calls to gitbase
 - [ ] [P1] gitbase interoperability with 3rd party tools
+    - [ ] Make compatible with MySQL workbench
```
Comment (mcuadros) [inline README.md]: I think that this a very concrete tasks, first we need to have a design meeting for this.

File: README.md [hunk 2]
```diff
@@ -5,10 +5,18 @@
 ### gitbase & engine
 
 - [ ] Improve gitbase performance
+    - [ ] Improve gitbase indexes
```
Comment (mcuadros) [inline README.md]: But This aren't only this the improvements we need to do for this Q

Comment (ajnavarro) [reply README.md]: Should we add here go-git performance improvements?

Comment (mcuadros) [reply README.md]: This will need a proper broke up after a design meeting, so maybe doesn't make sense write the KR now

**Pull request merged**